#!/bin/bash

PATCH="wxGTK-autoprofile.patch"
for i in x11-libs/wxGTK/wxGTK-*.ebuild; do
	cat > ${PATCH} << EOF
--- a/${i}	2013-10-06 11:31:35.000000000 +0200
+++ b/${i}	2013-09-18 23:44:22.652624338 +0200
@@ -170,10 +167,21 @@
 }
 
 pkg_postinst() {
 	has_version app-admin/eselect-wxwidgets \
 		&& eselect wxwidgets update
+
+	if [[ -e "\${ROOT}"/usr/lib/wx/config ]] ; then
+		local wxwidgets=( \$(find -H "\${ROOT}"/usr/lib/wx/config/* -printf "%f " 2> /dev/null) )
+		if [[ ! -z "\${wxwidgets[@]}" && "\${#wxwidgets[@]}" == 1 ]] ; then
+			eselect wxwidgets set  "\${wxwidgets[0]}"
+			echo
+			elog "Portage detected that your system has only one wxWidgets profile."
+			elog "Your systems wxWidgets profile is now set to \${wxwidgets[0]}"
+			echo
+		fi
+	fi
 }
 
 pkg_postrm() {
 	has_version app-admin/eselect-wxwidgets \
 		&& eselect wxwidgets update
EOF
	git apply ${PATCH}
	rm ${PATCH}
done
